scipy/stats/_continuous_distns.py

```diff
@@ -4659,6 +4659,38 @@ def _entropy(self):
         # https://en.wikipedia.org/wiki/Logistic_distribution
         return 2.0
 
+    def fit(self, data, *args, **kwds):
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
+
+        # if user has provided `floc` or `fscale`, fall back on super fit
+        # method. This scenario is not suitable for solving a system of
+        # equations
+        if floc is not None or fscale is not None:
+            return super(logistic_gen, self).fit(data, *args, **kwds)
+
+        # rv_continuous provided guesses
+        loc, scale = self._fitstart(data)
+        # account for user provided guesses
+        loc = kwds.pop('loc', loc)
+        scale = kwds.pop('scale', scale)
+
+        # the maximum likelihood estimators `a` and `b` of the location and
+        # scale parameters are roots of the two equations described in `func`.
+        # Source: Statistical Distributions, 3rd Edition. Evans, Hastings, and
+        # Peacock (2000), Page 130
+        def func(input, data):
+            a, b = input
+            n = len(data)
+            x1 = np.sum(np.exp((data - a) / b) /
+                        (1 + np.exp((data - a) / b))) - n / 2
+            x2 = np.sum(((data - a) / b) *
+                        ((np.exp((data - a) / b) - 1) /
+                         (np.exp((data - a) / b) + 1))) - n
+            return x1, x2
+
+        return tuple(optimize.root(func, (loc, scale), args=(data,)).x)
```

Comment:
does this not fail like the other method does?

Reply:
They are both victim to the same edge case, `data=np.ones(100)`. I'll add the fix here after we decide what to do in gumbel_r.

Reply:
I think we can consider the handling of an input that is exactly constant as an issue that will cause problems in all the distributions.  Even distributions that seem to work (i.e. return a result with no warnings or errors) are probably not giving a good result.  We can defer discussion of that to a separate issue, and get this in as is.